oop_student_report_project.py

```diff
@@ -0,0 +1,180 @@
+    percentage, blood group, address, and parent information.
+
+    Attributes:
+        name (str): The student's full name.
+        roll_number (str): The student's roll number.
+        m_marks (int): Marks obtained in mathematics.
+        p_marks (int): Marks obtained in physics.
+        c_marks (int): Marks obtained in chemistry.
+        eng_marks (int): Marks obtained in English.
+        percent (float): Overall percentage calculated as the average of subject marks.
+        b_group (str): Blood group of the student.
+        address (str): Address of the student.
+        m_name (str): Mother's name.
+        f_name (str): Father's name.
+        email (str): Email address of the student.
+
+    Methods:
+        add_student(): Allows user input to set the attributes for a student instance.
+    """
+     def __init__(self):
+        self.name = None
+        self.roll_number = None
+        self.m_marks = None
+        self.p_marks = None
+        self.c_marks = None
+        self.eng_marks = None
+        self.percent = None
+        self.b_group = None
+        self.address = None
+        self.m_name = None
+        self.f_name = None
+        self.email = None
+
+
+     def add_student(self):
+        """
+        Prompts the user to input details for a student and calculates the overall percentage.
+        """
+
+        self.name = input("\nEnter the student's name: ")
+        self.roll_number = input("Enter the student's roll no: ")
+        self.m_marks = int(input("Enter the student's maths marks: "))
+        self.p_marks = int(input("Enter the student's physics marks: "))
+        self.c_marks = int(input("Enter the student's chemistry marks: "))
+        self.eng_marks = int(input("Enter the student's english marks: "))
+        self.percent = round(((self.m_marks + self.p_marks + self.c_marks + self.eng_marks)/5), 2)
+        self.b_group = input("Enter the student's blood group: ")
+        self.address = input("Enter the student's address: ")
+        self.m_name = input("Enter the student's mother's name: ")
+        self.f_name = input("Enter the student's father's name: ")
+        self.email = input("Enter the student's email: ")
+
+def add_info(class_info, student_count):
+    """
+    Add student information to the class details.
+
+    Args:
+    class_details (dict): Dictionary containing student information.
+    num_students (int): Current number of students.
+
+    Returns: (tuple: Updated class_details and num_students.)
+    """
+    student = Student()
+    student.add_student()
+    class_info[student.roll_number] = student
+    student_count += 1
+    return class_info,student_count
+
+def generate_report(total_students, class_data):
+    """
+    Generate a report based on student information.
+
+    Args:
+    num_students (int): Current number of students.
+    class_details (dict): Dictionary containing student information.
```

Comment:
```suggestion
        num_students (int): Current number of students.
        class_details (dict): Dictionary containing student information.
```
Again I don't see these two arguments anywhere, please rename it.

Reply:
done